### PR TITLE
Support of enforced AWS MFA

### PR DESCRIPTION
**Pull request by igorkotof**
- Makefile requires to have AWS_MFA_ENABLED variable (true or false)
- After execution of local script for getting MFA session token you will be able to use ICMK targets which use AWS CLI

### CODE REVIEW

**Comment by igorkotof** [comment]
[ICMK support of enforced MFA](https://trello.com/c/zBkpuU9F/365-icmk-support-of-enforced-mfa)

**Review by kobrikx — APPROVED** [review]
LGTM

File: aws/main.mk [hunk 1]
```diff
@@ -4,27 +4,74 @@ ifndef AWS_REGION
...
+export AWS_ACCESS_KEY_ID=$(shell echo $(MFA_AWS_ACCESS_KEY_VALUE))
+export AWS_SECRET_ACCESS_KEY=$(shell echo $(MFA_AWS_SECRET_ACCESS_KEY_VALUE))
+export AWS_SESSION_TOKEN=$(shell echo $(MFA_AWS_SESSION_TOKEN_VALUE))
+endif
 # Macroses
 ########################################################################################################################
 # We don't check for AWS_PROFILE, but instead we assume the profile name.
 # You can override it, although it's recommended to have a profile per environment in your ~/.aws/credentials
 AWS_PROFILE ?= $(NAMESPACE)-$(ENV)
+AWS_CLI_PROFILE ?= $(shell echo $$(if [ "$(AWS_MFA_ENABLED)" = "true" ]; then echo ""; else echo "--profile $(AWS_PROFILE)"; fi))
```
Comment (AutomationD) [inline aws/main.mk:17]: Do you think this should be reflecting the `MFA` nature? like `AWS_PROFILE_MFA`?

File: aws/main.mk [hunk 2]
```diff
@@ -4,27 +4,74 @@ ifndef AWS_REGION
...
 AWS_USER ?= $(shell [ -f ~/.aws/credentials ] && $(AWS) iam get-user | $(JQ) -r ".User.UserName")
 AWS_ACCOUNT ?= $(shell [ -f ~/.aws/credentials ] && $(AWS) sts get-caller-identity | $(JQ) -r '.Account' || echo "nil" )
 
 AWS_DEV_ENV_NAME ?= $(shell [ -f ~/.aws/credentials ] && $(AWS) iam list-user-tags --user-name $(AWS_USER) | ( $(JQ) -e -r '.Tags[] | select(.Key == "devEnvironmentName").Value') || echo "$(ENV) (User env is not configured)")
 
+# AWS MFA
+MFA_DEVICE_ARN ?=
```
Comment (AutomationD) [inline aws/main.mk:24]: Should `MFA_*` be more like `AWS_IAM_MFA_*` or `AWS_MFA_*`?

Comment (AutomationD) [reply aws/main.mk:24]: Or maybe `MFA_TOKEN_CODE` should become `MFA_AWS_TOKEN_CODE`?

**Review by AutomationD — CHANGES_REQUESTED** [review]

File: aws/main.mk (hunk 1 above)
Comment (igorkotof) [reply aws/main.mk:17]: @AutomationD I do not think so since if we use MFA - there is no needs in AWS Profile, thoughts?

File: aws/main.mk (hunk 2 above)
Comment (igorkotof) [reply aws/main.mk]: agree, fixed

File: aws/main.mk (hunk 1 above)
Comment (AutomationD) [reply aws/main.mk:17]: Okay, sounds good. Let's keep it as is.

**Review by AutomationD — APPROVED** [review]

**Pull request merged**